src/charm.py

```diff
@@ -121,8 +123,8 @@ def _on_start(self, event: ops.charm.StartEvent) -> None:
                 self.unit.status = BlockedStatus("couldn't start MongoDB")
                 return
 
+        # open TCP port
```

Comment:
You can probably drop this on account of the sensible method name you're about to call ;-)